cicecore/shared/ice_fileunits.F90

```diff
@@ -239,7 +243,8 @@ subroutine release_all_fileunits
          call release_fileunit(nu_rst_pointer)
          call release_fileunit(nu_history)
          call release_fileunit(nu_hdr)
-         if (nu_diag /= ice_stdout) call release_fileunit(nu_diag)
+         ! CESM can have negative unit numbers
+         if (nu_diag > 0 .and. nu_diag /= ice_stdout) call release_fileunit(nu_diag)
```

Comment:
Maybe rather than change this logic, it should just be

#ifndef CESMCOUPLED
  if (nu_diag /= ice_stdout) call release_fileunit(nu_diag)
#endif

Reply:
We still need to make sure not to release the negative file units here. However, I am fine with wrapping in a CESMCOUPLED.